StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift

```diff
@@ -90,8 +92,55 @@ extension EmbeddedPaymentElement: EmbeddedPaymentMethodsViewDelegate {
         delegate?.embeddedPaymentElementDidUpdateHeight(embeddedPaymentElement: self)
     }
 
-    func selectionDidUpdate() {
-        delegate?.embeddedPaymentElementDidUpdatePaymentOption(embeddedPaymentElement: self)
+    func selectionWasTapped(didUpdate: Bool) {
+        // Defer notifying delegate until we are about to exit this function to give time to either allow the new payment option to come from
+        // The new `EmbeddedFormViewController` instance if needed
+        defer {
+            if didUpdate {
+                delegate?.embeddedPaymentElementDidUpdatePaymentOption(embeddedPaymentElement: self)
+            }
+        }
+        
+        guard case let .new(paymentMethodType) = embeddedPaymentMethodsView.selection else {
+            // This can occur when selection is being reset to nothing selected, so don't assert.
+            return
+        }
+
+        guard let presentingVC = presentingViewController else {
+            assertionFailure("Presenting view controller not found, set EmbeddedPaymentElement.presentingViewController.")
+            return
+        }
+
+        let formViewController = EmbeddedFormViewController(
+            configuration: configuration,
+            intent: intent,
+            elementsSession: elementsSession,
+            savedPaymentMethods: savedPaymentMethods,
+            paymentMethodType: paymentMethodType,
+            previousPaymentOption: self.formViewController?.previousPaymentOption,
+            analyticsHelper: analyticsHelper,
+            formCache: formCache
+        )
+        formViewController.delegate = self
+
+        // Only show forms that require user input
+        guard formViewController.collectsUserInput else {
+            self.formViewController = nil  // Clear out any previous form view controller to update self._paymentOption
+            return
+        }
+
+        let bottomSheet = BottomSheetViewController(
+            contentViewController: formViewController,
+            appearance: configuration.appearance,
+            isTestMode: configuration.apiClient.isTestmode,
+            didCancelNative3DS2: {
+                stpAssertionFailure("3DS2 was triggered unexpectedly")
+            }
+        )
```

Comment:
nit: since i'm seeing duplicate code in 3 spots now... maybe should add a function:
```
    func bottomSheetController(with viewController: BottomSheetContentViewController) -> BottomSheetViewController {
        return BottomSheetViewController(contentViewController: viewController,
                                         appearance: configuration.appearance,
                                         isTestMode: configuration.apiClient.isTestmode,
                                         didCancelNative3DS2: {
            stpAssertionFailure("3DS2 was triggered unexpectedly")
        })
    }
```